projects/bootloader/src/ping.c

```diff
@@ -0,0 +1,31 @@
+#include "ping.h"
+
+#include <stdint.h>
+
+#include "bootloader_can.h"
+#include "bootloader_datagram_defs.h"
+#include "can_datagram.h"
+#include "config.h"
+#include "dispatcher.h"
+
+// response config datagram setup
+static uint8_t s_board_id;
+static CanDatagramTxConfig s_response_config = {
+  .dgram_type = BOOTLOADER_DATAGRAM_PING_RESPONSE,
+  .destination_nodes_len = 0,  // client listens to all datagrams
+  .destination_nodes = NULL,
+  .data_len = 1,
+  .data = &s_board_id,
+  .tx_cb = bootloader_can_transmit,
+  .tx_cmpl_cb = tx_cmpl_cb,
+};
+
+static StatusCode prv_ping_response(uint8_t data[2048], uint16_t data_len, void *context) {
```

Comment:
Unless there is a benefit to using a formal array parameter here that I am missing, I would make this a uint8_t*

Reply:
will be fixed in the next commit.